modules/programs/zsh.nix

```diff
@@ -368,6 +379,13 @@ in
         description = "Options related to zsh-history-substring-search.";
       };
 
+      autosuggestion = mkOption {
```

Comment:
I think you can skip the submodule. in this case it should be possible to simply do something like

```
autosuggestion = {
  enable = mkOption …;
  highlight = mkOption …;
};

Reply:
ha that's how I had first implemented it than I saw the highlighting module and decided to copy whatthey did. I will revert that.

Reply:
Yeah, we should probably also simplify the highlighting options in the same way. The submodule looks unnecessary.